actionpack/lib/action_controller/metal/renderers.rb

```diff
@@ -89,6 +123,68 @@ def self.remove(key)
       remove_method(method_name) if method_defined?(method_name)
     end
 
+    # TODO: update text
+    # Serializers define a method called within a renderer specific to
+    # transforming the object into a mime-compatible type.
+    # See <tt>Renderers.add</tt>
+    #
+    # The separation of serialization from rendering allows
+    # composing the Renderer behavior of two methods, e.g.
+    # +_render_with_renderer_json+ and +_serialize_with_serializer_json+,
+    # rather than requiring one to define a method +_render_with_renderer_json+
+    # in a subclass and optionally call super on it.
+    #
+    # A principal benefit of this approach is that it promotes serialization of an object
+    # to a clearly-defined public interface, rather than requiring one to understand that
+    # calling, e.g. +render json: object+ calls +_render_to_body_with_renderer(options)+
+    # which calls +_render_with_renderer_#{key}+ where key is +json+, which is the method
+    # defined by calling +ActionController::Renderers.add :json+.
+    #
+    # Example usage:
+    #
+    # Prior to the introduction of SERIALIZERS, customizing JSON serialization would
+    # have relied upon defining +_render_with_renderer_json+ (or +_render_option_json+,  pre-4.2 )
+    # in the controller, and calling +super+ on
+    # the serialized object. Now, one need only call +ActionController.remove_serializer :json+ and
+    # define a new serializer with +ActionController.add_serializer json do |json, options| end+.
+    # There's no longer a need to add controller methods to define custom serializers.
+    #
+    # Pretty-printing JSON can be implemented by replacing the JSON serializer:
+    #
+    #   ActionController::Renderers.remove_serializer :json
+    #   ActionController::Renderers.add_serializer :json do |json, options|
+    #     return json if json.is_a?(String)
+    #
+    #     json = json.as_json(options) if json.respond_to?(:as_json)
+    #     json = JSON.pretty_generate(json, options)
+    #   end
+    #
+    #   Or in the controller:
+    #
+    #   class TheController < ApplicationController
+    #     serializing json: ->(json, options) do
+    #       return json if json.is_a?(String)
+    #
+    #       json = json.as_json(options) if json.respond_to?(:as_json)
+    #       JSON.pretty_generate(json, options)
+    #     end
+    #   end
+    #
+    # See https://groups.google.com/forum/#!topic/rubyonrails-core/K8t4-DZ_DkQ/discussion for
+    # more background information.
+    def self.add_serializer(key, &block)
+      SERIALIZERS[key.to_sym] = block
+    end
```

Comment:
- [ ] TODO: keep this interface? Since `serializing` is only availabe in the controller class, we need something at the module level for config, I think.  baby steps